src/main/java/org/broadinstitute/hellbender/engine/dataflow/datasources/VariantShard.java

```diff
@@ -0,0 +1,98 @@
+package org.broadinstitute.hellbender.engine.dataflow.datasources;
+
+import com.google.cloud.dataflow.sdk.coders.*;
+import com.google.cloud.dataflow.sdk.values.KV;
+import htsjdk.samtools.util.Locatable;
+import org.broadinstitute.hellbender.utils.variant.Variant;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * VariantShard is section of the genome that's used for sharding work for pairing things with
+ * variants. The primary use case is pairing reads with overlapping variants.
+ * This class designed to be simple a data-storage class with related static utilities.
+ */
+public final class VariantShard {
+    private final int shardNumber;
+    private final String contig;
+    static public final int VARIANT_SHARDSIZE = 100000; // This value is likely to be tweaked.
+
+    public VariantShard(int shardNumber, String contig) {
+        this.shardNumber = shardNumber;
+        if (contig == null) {
```

Comment:
Should there be a check that this is less than VARIANT_SHARDSIZE? Or is there another relationship between the two?